torch_xla/csrc/device.cpp

```diff
@@ -9,39 +9,49 @@
 namespace torch_xla {
 namespace {
 
-thread_local absl::optional<torch::lazy::BackendDevice> g_current_device;
+// This is set when any device is initialized, so to prevent using non-virtual
+// device and virtual device together.
+static bool spmd_config_is_locked = false;
+
+}  // namespace
+
+std::string DeviceType::XlaDeviceTypeToString(XlaDeviceType hw_type) {
+  XLA_CHECK(hw_type != XlaDeviceType::PLUGIN) << "PLUGIN type name unknown";
```

Comment:
when will this function be called? Are we not expceting this to be called on plug in?

Reply:
This function is used when constructing a device from the `XlaDeviceType` enum type rather than the device name in a string. E.g. here:https://github.com/pytorch/xla/blob/9a4ef68a1ff5b53c1f793d9b2370836a29dc93dc/torch_xla/csrc/xla_backend_impl.cpp#L183-L184

If all we get is the placeholder `XlaDeviceType::PLUGIN` type, then we don't know the actual `platform_name` of the device type for `toString`. I think this is a relatively rare case in our code, so I can try to remove it. I have a more ambitious refactoring effort going in https://github.com/pytorch/xla/pull/6261.